### PR TITLE
Straighten permissions for `Individual.hasTwoFactorAuth`

### PR DESCRIPTION
**Pull request by Betree**

### CODE REVIEW

File: server/graphql/v2/object/Individual.js [hunk 1]
```diff
@@ -102,7 +102,9 @@ export const GraphQLIndividual = new GraphQLObjectType({
         type: GraphQLBoolean,
         async resolve(collective, args, req) {
           const user = await req.loaders.User.byCollectiveId.load(collective.id);
-          return twoFactorAuthLib.userHasTwoFactorAuthEnabled(user);
+          if (req.remoteUser.id === user.id) {
```
Comment (znarf) [inline server/graphql/v2/object/Individual.js:105]: There is no guarantee that `req.remoteUser` is set right?

Comment (znarf) [inline server/graphql/v2/object/Individual.js:105]:
```suggestion
          if (req.remoteUser?.id === user.id) {
```

**Review by znarf — CHANGES_REQUESTED** [review]
See my feedback, unless I'm mistaken.

File: server/graphql/v2/object/Individual.js (hunk 1 above)
Comment (Betree) [reply server/graphql/v2/object/Individual.js:105]: All the queries where we fetch this info are logged in, but you're right - there's no guarantee of it. I copied the snippet from V1's `hasTwoFactorAuth`, I'll fix it there and in `hasPassword` that uses the same logic too.

**Review by znarf — APPROVED** [review]

**Pull request merged**